src/app/application.cpp

```diff
@@ -512,6 +512,7 @@ void Application::initializeTranslation()
     Preferences* const pref = Preferences::instance();
     // Load translation
     QString localeStr = pref->getLocale();
+    QLocale::setDefault(QLocale(localeStr));
```

Comment:
>Don't we need to reset this if user changes translation during runtime?

`the default locale should be set at application startup, before any non-GUI threads are created.`
see: http://doc.qt.io/qt-4.8/qlocale.html#setDefault

>Also will this affect the printing of thousand/decimal separators? I think the user would prefer his system defined separators.

Probably, but if a user wants to change the language (different to their system), I would say that user should also accommodate to the new decimal separator.